simple-wp-helpdesk/includes/class-email.php

```
@@ -159,6 +169,158 @@ function swh_wrap_html_email( $body, $attachments = array() ) {
+ * @since 3.0.0
+ * @param WP_REST_Request $request The REST request object.
+ * @return WP_REST_Response|WP_Error
+ */
+function swh_handle_inbound_email( $request ) {
+	// Optional Bearer token authentication.
+	$secret = swh_get_string_option( 'swh_inbound_secret' );
+	if ( $secret ) {
+		$auth   = $request->get_header( 'Authorization' );
+		$bearer = is_string( $auth ) ? trim( str_replace( 'Bearer', '', $auth ) ) : '';
+		if ( ! hash_equals( $secret, $bearer ) ) {
+			return new WP_Error( 'swh_unauthorized', __( 'Unauthorized.', 'simple-wp-helpdesk' ), array( 'status' => 401 ) );
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '190', column '15'). (undefined)

(MissingImport)